pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/Vector.java

```diff
@@ -0,0 +1,261 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.data.readers;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.util.Arrays;
+
+
+public class Vector implements Comparable<Vector> {
+  public enum VectorType {
+    FLOAT, INT
```

Comment:
Do you think FLOAT is enough? Where INT could be used?

Reply:
Let's also add BYTE

Reply:
Plus one on this, can we consider to support or at leas be forward compatible with float16?
https://github.com/milvus-io/milvus/issues/22837
https://github.com/Writer-X/milvus/commit/39d8e2634708f5d76bf1195d15ccd965eb03024b

Reply:
JDK 20 has native float16 support.
For lower JDK version, we have to wrap a `BYTE[2]` or `SHORT` as a `FLOAT16`.